src/gc/transaction_level_gc_manager.cpp

```diff
@@ -135,27 +138,28 @@ int TransactionLevelGCManager::Unlink(const int &thread_id, const cid_t &max_cid
     }
   }  // end for
 
-  auto safe_max_cid = concurrency::TransactionManagerFactory::GetInstance().GetNextCommitId();
+  eid_t safe_expired_eid = concurrency::EpochManagerFactory::GetInstance().GetNextEpochId();
```

Comment:
May be we should add a COMPILER FENCE before acquiring this epoch id? We need to reach a barrier here.